sdks/java/core/src/main/java/org/apache/beam/sdk/io/BigQueryIO.java

```diff
@@ -550,13 +550,13 @@ private static void dryRunQuery(BigQueryOptions options, String query) {
         final BigQueryServices bqServices = getBigQueryServices();
 
         final String extractDestinationDir;
-        String tempLocation = bqOptions.getTempLocation();
+        String bqTempLocation = bqOptions.getBigQueryTempLocation();
```

Comment:
I think that this is not necessary. Would rather just default to `GcpTempLocation` and let users additionally override the temp location for each individual BQ read or write.